h2o-core/src/main/java/hex/Model.java

```diff
@@ -399,6 +399,26 @@ public long getOrMakeRealSeed(){
     /** @return the validation frame instance, or null
      *  if a validation frame was not specified */
     public final Frame valid() { return _valid==null ? null : _valid.get(); }
+
+    /**
+     * @return all Frames used by these params
+     */
+    public final Map<String, Frame> getAllFrames() {
+      Map<String, Frame> framesMap = new HashMap<>();
+      collectAllFrames(framesMap);
+      return framesMap;
+    }
+    
+    protected void collectAllFrames(Map<String, Frame> map) {
+      collectFrame(map, "train", _train);
```

Comment:
Should the type of the frame be an enum ? Possibly algo-specific ? Java doesn't support enum inheritance, could be done with constants in `FrameType` class or something like that. Just thinking out loud.